auto_editor/StructuredProjectSource.py

```diff
@@ -41,10 +41,28 @@ def get_all_dpct_file_paths(self) -> List[str]:
         # Todo: use same logic here as in PreAnalyser. Currently nested files aren't discovered.
 
         paths = []
-
-        for file_path in os.listdir(self.dpct_root):
-            if file_path.endswith(".dp.cpp") or file_path.endswith(".dp.hpp"):
-                paths.append(file_path)
+        test_path = str(self.dpct_root.stem) #'test_project'
+        print('path name', test_path)
+        num = 0
+        # for file_path in os.listdir(self.dpct_root):
+        #     if file_path.endswith(".dp.cpp") or file_path.endswith(".dp.hpp"):
+        #         paths.append(file_path)
+
+        for file_dpcpp in self.dpct_root.rglob('*.dp.cpp'):
+            temp1 = file_dpcpp.parts
+            if test_path in temp1:
+                num = temp1.index(test_path)
+            dpcpp_path = '/'.join(temp1[num+1:])
+            paths.append(dpcpp_path)
+
+        for file_dphpp in self.dpct_root.rglob('*.dp.hpp'):
+            temp2 = file_dphpp.parts
```

Comment:
what is temp2? For readability, it would be good to have a more self-explanatory name. Or just delete this line and fdo `if test_path in file_dphpp.parts:` below